.github/workflows/nightly_tests.yaml

```diff
@@ -0,0 +1,32 @@
+name: Go nightly build
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'
+jobs:
+  test_client:
+    runs-on: ${{ matrix.os }}
+    name: Run tests of master on ${{ matrix.os }} with Go ${{ matrix.go_version }}
+    strategy:
+        matrix:
+            os: [ ubuntu-latest ]
+            go_version: [ 1.17, 1.18 ]
+        fail-fast: false
+    steps:     
+    
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+          
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go_version }}
+          
+      - name: Checkout code
+        uses: actions/checkout@v2
+          
+      - name: Test
+        run: |
+          ./rc.sh start
+          env MEMBER_COUNT=3 make test-cover
```

Comment:
It would be better to specify Hazelcast versions explicitly:
```
env HZ_VERSION=5.1 ./rc.sh start
env HZ_VERSION=5.1 MEMBER_COUNT=3 make test-cover
```

Reply:
Added.
It is the same in the coverage report workflow, we should change it there I think but before changing it we can consider to use always latest release of the server. WDYT?

Reply:
Thanks. We should change that as well.